source/common/quic/quic_filter_manager_connection_impl.cc

```diff
@@ -157,6 +157,7 @@ void QuicFilterManagerConnectionImpl::onConnectionCloseEvent(
     raiseConnectionEvent(source == quic::ConnectionCloseSource::FROM_PEER
                              ? Network::ConnectionEvent::RemoteClose
                              : Network::ConnectionEvent::LocalClose);
+    quic_connection_ = nullptr;
```

Comment:
I suspect quic_connection_ != nullptr is always true after moving it here. Can you change to assert and see if all tests pass?

Reply:
yep, passes, PTAL